modules/sor/sorV2/lib/poolsV3/buffer/bufferPool.ts

```diff
@@ -0,0 +1,115 @@
+import { Address, Hex } from 'viem';
+import { MAX_UINT256, SwapKind, Token, TokenAmount } from '@balancer/sdk';
+import { BufferState, Vault } from '@balancer-labs/balancer-maths';
+
+import { BasePoolV3 } from '../../poolsV2/basePool';
+import { BasePoolToken } from '../../poolsV2/basePoolToken';
+import { Erc4626PoolToken } from '../../poolsV2/erc4626PoolToken';
+
```

Comment:
Would be helpful to add a bit of documentation here explaining what how the BufferPool and ERC4626 relate.